web/src/pages/user-profile.tsx

```diff
@@ -1,16 +1,66 @@
-import { Box, Flex, Text } from '@chakra-ui/react';
-import { ENDPOINT_CONFIGS, GetUserRequest, GetUserResponse, withParams } from '@codersquare/shared';
+import { Button, Flex, Input, InputGroup, InputLeftAddon, Text } from '@chakra-ui/react';
+import {
+  ENDPOINT_CONFIGS,
+  GetUserRequest,
+  GetUserResponse,
+  UpdateCurrentUserRequest,
+  UpdateCurrentUserResponse,
+  withParams,
+} from '@codersquare/shared';
 import { useQuery } from '@tanstack/react-query';
+import { FormEvent, useCallback, useContext, useState } from 'react';
 import { useParams } from 'react-router-dom';
 
-import { callEndpoint } from '../fetch';
+import { UserCtx } from '../App';
+import { ApiError, callEndpoint } from '../fetch';
 
 export const UserProfile = () => {
   const { id } = useParams();
-  const { data, error, isLoading } = useQuery([`getuser${id}`], () =>
-    callEndpoint<GetUserRequest, GetUserResponse>(withParams(ENDPOINT_CONFIGS.getUser, id!))
+  const [userName, setUserName] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [editingMode, setEditingMode] = useState(false);
+  const [userUpdateError, setUserUpdateError] = useState('');
+  const { currentUser } = useContext(UserCtx);
+  const isOwnProfile = id === currentUser.id;
+
+  const { data, error, isLoading } = useQuery(
+    [`getuser${id}`],
+    () => callEndpoint<GetUserRequest, GetUserResponse>(withParams(ENDPOINT_CONFIGS.getUser, id!)),
+    {
+      onSuccess: data => {
+        setUserName(data.userName);
+        setFirstName(data.firstName || '');
+        setLastName(data.lastName || '');
+      },
+    }
+  );
+
+  const updateCurrentUser = useCallback(
+    async (e: FormEvent | MouseEvent) => {
+      e.preventDefault();
+      try {
+        await callEndpoint<UpdateCurrentUserRequest, UpdateCurrentUserResponse>(
+          ENDPOINT_CONFIGS.updateCurrentUser,
+          {
+            userName,
+            firstName,
+            lastName,
+          }
+        );
+        window.location.reload();
```

Comment:
instead of a full reload, we could just call the `setCurrentUserGlobally` callback from the context?

Reply:
also `setUserUpdateError('')` in case of success here.